README.md

```diff
@@ -20,127 +19,304 @@ let thirdNumber = numbers[2]
+
+```swift
+var toMakeTea = [
+    "Boil Water",
+    "Add tea bag to cup",
+    "Wait ten minutes",
+    "Add Milk",
+    "Drink"
+]
+
+print("1\. \(toMakeTea[0])")
+print("2\. \(toMakeTea[1])")
+// and so on
+
+// prints "1\. Boil Water"
+// prints "2\. Add tea bag to cup"
+// etc
+```
+
+This way of accomplishing the task is incredibly tedious. Knowing how a `for-in` loop works, you can iterate over each value in the array of strings. You also the index of the value in the array because you want to display numbered instructions.
+
+If steps need to be added, removed, or changed at a particular index in the array, then you also need to know what the index of the value is. When you have a simple list of unchanging values, then maybe this is simple, but with a larger list processed in a loop, there is another way, the `.enumerate` method that you call on an array variable.
+
+```swift
+
+for (index, step) in toMakeTea.enumerate() {
+    print("\(index + 1). \(step)")
+}
+
+// 1\. Boil Water
+// 2\. Add tea bag to cup
+// 3\. Wait ten minutes
+// 4\. Add Milk
+// 5\. Drink
+```
+
+In this example you first create an array of the steps to make tea (and yes, there are intentional mistakes!) and add a couple of extra things to the `for in` loop. First inside the brackets you add variables to hold the index and value variables, the names of these are up to you. Using `index` is of course a useful name, but you can call it whatever you want. At the end of the array variable you want to iterate through, add the `.enumerate()` method. Now inside the loop you have access to the value in the current iteration as well as it's index in the array. Above, you add '1' to the index value as unlike computers, humans don't count from 0!
+
+## Check an array has values
+
+Before undertaking any further actions, it's a good idea to check the array you would like to work with actually has any values in it. Of course there's a method to help, `.isEmpty`.
+
+```swift
+if toMakeTea.isEmpty {
+   print("No tea yet :(")
+} else {
+   print("Ready to make tea! :)")
+}
+
+// prints "No tea yet :("
+```
+
+This method checks to see if the array contains any values, but the array still needs to actually exist (i.e. it's you have initiated it) in the first place.
+
+![Array not Initialized](not-init.png)
+
+It might be a good idea to add this check to your array before any other functionality we look at in the rest of the lesson, wrapping it in `if else` statements.
+
+You can reverse a logical check with the `!` symbol:
+
+```swift
+if !toMakeTea.isEmpty {
+    print("Ready to make tea! :)")
+} else {
+    print("No tea yet :(")
+}
+```
+
+This may seem confusing syntax, but it may feel more natural to start with a preferred or default status and then handle the alternative or fallback status.
+
+## Array Size
+
+There are some methods that help you access certain properties of an array, one is to find the size of an array, or the number of values it contains.
+
```

Comment:
@JimCampagno Can we add a sentence here that makes this more story like? "What if we wanted to know the number of steps it took to make tea? The elements in our toMakeTea array are those steps and while we can just count them, there is a method that will  find the size of an array....."